client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/AddIns/DockPanes/StreetSmart.cs

```diff
@@ -613,48 +613,51 @@ await QueuedTask.Run(() =>
 
         try
         {
-          EventLog.Write(EventLog.EventType.Information, $"Street Smart: (StreetSmart.cs) (OpenImageAsync) Open image: {toOpen}");
-          IList<IViewer> viewers = await Api.Open(toOpen, viewerOptions);
-
-          if (Nearest && _toRestartImages.Count == 0 && toOpen == _location && point != null)
+          if (!string.IsNullOrEmpty(toOpen))
           {
-            if (_crossCheck == null)
-            {
-              _crossCheck = new CrossCheck();
-            }
+            EventLog.Write(EventLog.EventType.Information, $"Street Smart: (StreetSmart.cs) (OpenImageAsync) Open image: {toOpen}");
+            IList<IViewer> viewers = await Api.Open(toOpen, viewerOptions);
```

Comment:
move API call closer to place of usage

Reply:
This API call cannot be moved lower because it opens the image at this point. The if block below is executed when a user clicks somewhere on the map where there isn’t a specific point with a method Ctrl + click